### ISSUE TITLE
request time out error in synchronization time  -SalesforceMobileSDK-CordovaPlugin Ionic App 

### ISSUE THREAD

**Issue by asif986** [forcedotcom/SalesforceMobileSDK-CordovaPlugin#637]
Hi 
We are using [SalesforceMobileSDK-CordovaPlugin] to develop hybrid ios app. We are making request through forcejs which uses network plugin .

sometime, when app sync is start we received **Request Time out** error in SalesforceMobileSDK-CordovaPlugin.

Can anyone help me.

SalesforceMobileSDK-CordovaPlugin 6.2.0
cordova-ios - 6.0.0
Ionic - 4.0.0

**Comment by wmathurin** [comment]
You cannot doing from JavaScript but you should be able to change the timeout setting for the NSURLSession using the [SFNetwork](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFNetwork.h). You can even customize the creation of NSURLSessions if you were to implement the SFNetworkManaging protocol. However this change will apply for all requests.

**Comment by asif986** [comment]
Thank you for your prompt response @wmathurin 
We are using  **SalesforceMobileSDK-CordovaPlugin 6.2.0**, and the file you mention in (SFNetwork) does not exist.

![NetworkSession](https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin/assets/69723211/cc2057ff-7236-4589-808c-699219be46ee)

**Cordova plugins**
![image](https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin/assets/69723211/b8b81978-1c68-4e2a-b53c-edf9ecbd3a16)

**Comment by wmathurin** [comment]
MSDK 6.2.0 was released in June 2018 - almost 6 years ago. Have you considered upgrading the version of the Mobile SDK you app uses?

**Comment by asif986** [comment]
@wmathurin Yes, We are aware the version we are using is quite old, but we have our ORG restrictions which is the reason we couldn't update the version.

It would be great if you could share any to add this SFNetwork file in SDK 6.2.0 

As per I know, we are adding the header files in **header folder** under the iOS platform. So can we do the same for SFNetwork?

![image](https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin/assets/69723211/895a7b27-5321-464a-8780-993e98e5e435)

**Comment by wmathurin** [comment]
Unfortunately we do not patch old versions of the Mobile SDK (especially 5 year old versions).
If you can change source files in your build why can't you upgrade the Mobile SDK?
